src/Constraint/RequestConstraintValidator.php

```diff
@@ -38,43 +39,45 @@ public function validate($value, Constraint $constraint): void
         $this->validateAttributes($constraint, $value);
     }
 
-    private function validateQuery(RequestConstraint $constraint, Request $value): void
+    private function validateQuery(RequestConstraint $constraint, Request $request): void
     {
         if ($constraint->query !== null) {
             $this->context->getValidator()
                 ->inContext($this->context)
                 ->atPath('[query]')
-                ->validate($value->query->all(), $constraint->query);
-        } elseif ($constraint->allowExtraFields === false && count($value->query) > 0) {
+                ->validate($request->query->all(), $constraint->query);
+        } elseif ($constraint->allowExtraFields === false && count($request->query) > 0) {
             $this->context->buildViolation($constraint->queryMessage)
                 ->atPath('[query]')
                 ->setCode($constraint::MISSING_QUERY_CONSTRAINT)
                 ->addViolation();
         }
     }
 
-    private function validateRequest(RequestConstraint $constraint, Request $value): void
+    private function validateRequest(RequestConstraint $constraint, Request $request): void
     {
         if ($constraint->request !== null) {
+            $data = in_array($request->getContentType(), ['json', 'jsonld'], true) ? $request->toArray() : $request->request->all();
```

Comment:
RequestConstraintValidator::validate / RequestConstraintValidator::validateRequest are not typehinted to throw Exceptions. Catch the toArray exception and make it fail the validation.